website/docs/plugins/provider.html.md

```diff
@@ -32,6 +32,44 @@ of this page will assume you're familiar with
 [plugin basics](/docs/plugins/basics.html) and that you already have
 a basic development environment setup.
 
+## Provider Plugin Codebases
+
+Provider plugins live outside of the Terraform core codebase in their own
+source code repositories. The official set of provider plugins released by
+Hashicorp (developed by both Hashicorp staff and community contributors)
+all live in repositories in
+[the `terraform-providers` organization](https://github.com/terraform-providers)
+on GitHub, but third-party plugins can be maintained in any source code
+repository.
+
+When developing a provider plugin, it is recommended to use a common `GOPATH`
+that includes both the core Terraform repository and the repositories of any
+providers being changed. This makes it easier to use a locally-built
+`terraform` executable and a set of locally-built provider plugins together
+without further configuration. For example, for the `template` provider:
+
+```
+$ go get github.com/hashicorp/terraform
+...
+$ go get github.com/terraform-providers/terraform-provider-template
+...
+$ go install github.com/hashicorp/terraform
```

Comment:
I think we can drop the `...` and install steps here. `go get` runs `install`, so the following paragraph is still true, but the "do something" `...` lines and the install steps just seem extra noise for the target developers which should know how to use the `go` tool, and possibly confusing for those who don't.